src/rewriting.jl

```diff
@@ -1,3 +1,43 @@
+"""
+    rewrite_from_left(u::AbstractWord, rewriting)
+Rewrites a word from left using `rewriting` object. The object must implement
+`rewrite_from_left!(v::AbstractWord, w::AbstractWord, rewriting)` to succesfully rewrite `u`.
+"""
+function rewrite_from_left(u::W, rewriting) where {W<:AbstractWord}
```

Comment:
Wouldn't it be sensible to declare additional type `RewritingObject` which would be a supertype of `AbstractRewritingSystem` and `AbstractAutomaton`?

Reply:
sure, why not; but I see little (no?) added value with this abstraction, except making the signature clearer;
I hope the docstring is clear enough?

On the other hand if someones `rewriting_object` is already a subtype of a distinct type hierarchy we'd effectively force to define a shim around it to be used here. That's far fetched, so that I wouldn't count this as an argument here though.